threedworld-mit/tdw
# Commands for rotating the directional light

# How to test

- Review the new API documentation--the distinction between HDRI skybox suns and the directional light might not be clear enough

Then...

1. In the `tdw` repo, `git fetch` and `git checkout directional_light`
2. Download [this build](https://www.dropbox.com/s/jai3gnd2crnrbia/TDW_Windows_directional_light.zip?dl=0)
3. Run [this controller](https://raw.githubusercontent.com/threedworld-mit/tdw/676b835ab858cd40ff31f142a6d73896027cb95d/Python/tdw/directional_light.py)
4. Run the build
5. Check images (the Python console where tell you where the images are)

# Changelog

### Command API

#### New Commands

| Command                            | Description                                                  |
| ---------------------------------- | ------------------------------------------------------------ |
| `rotate_directional_light_by`      | Rotate the directional light (the sun) by an angle and axis. |
| `reset_directional_light_rotation` | Reset the rotation of the directional light (the sun).       |

### Example Controllers

- Added: `directional_light.py`
